packages/core/src/config.ts

```
@@ -955,64 +959,113 @@ const composeEntryConfig = async (
-            // user should use copy to keep origin file or use another separate entry to deal this
-            let request: string = data.request;
-
-            const cssExternal = cssExternalHandler(
-              request,
-              callback,
-              jsExtension,
-              cssModulesAuto,
-              isStyleRedirect,
-            );
-
-            if (cssExternal !== false) {
-              return cssExternal;
+    resolvedJsRedirect: {
+      path: jsRedirectPath,
+      extension: jsRedirectExtension,
+    },
+    config: {
+      output: {
+        externals: [
+          async (data, callback) => {
+            const { request, getResolve, context, contextInfo } = data;
+            if (!request || !getResolve || !context || !contextInfo) {
+              return callback();
+            }
+
+            if (!resolver) {
+              resolver = (await getResolve()) as Resolver;
             }
 
-            if (request[0] === '.') {
-              const ext = extname(request);
+            // Issuer is not empty string when the module is imported by another module.
+            // Prevent from externalizing entry modules here.
+            if (contextInfo.issuer) {
+              let resolvedRequest: string = request;
+
+              const cssExternal = cssExternalHandler(
+                resolvedRequest,
+                callback,
+                jsExtension,
+                cssModulesAuto,
+                doesRedirectStyle,
+              );
+
+              if (cssExternal !== false) {
+                return cssExternal;
+              }
+
+              if (jsRedirectPath) {
+                try {
+                  resolvedRequest = await resolver(context, resolvedRequest);
+                } catch (e) {
+                  // Do nothing, fallthrough to other external matches.
+                  logger.debug(
+                    `Failed to resolve ${resolvedRequest} with resolver`,
```

should the error be printed to debug info？

---

I'm not sure which request failed to resolve. Bundleless is similar to transform, so regarding transform, we might be able to ignore the error for now.

---

IMO, #251 means: the rewritten path should be "resovled", which is the issue addressed in this PR. The error to omit here is that a module can't be resolved. They're slightly different.

---

@SoonIter What's your opinion?

---

I think we can first come up with a reasonable use case for this error here. If we can't find one, we can assume that there is no use case, and then we can directly throw an error.

---

Leave it here for now, we could change this in the future if needed.